### PR TITLE
Added cycle sort

### PR DESCRIPTION
**Pull request by gbrunofranco**

### CODE REVIEW

File: sorting/cycle_sort.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			item = tmp;
+		}
+	}	
+}
+int main()
+{
+	int size = 6;
+	int *numbers = malloc(size*sizeof(int));
+	printf("Insert %d unsorted numbers: \n", size);
+	for (int i = 0; i < size; ++i)
```
Comment (christianbender) [inline sorting/cycle_sort.c]: Write a function for printing the array.

File: sorting/cycle_sort.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			int tmp = numbers[pos];
+			numbers[pos] = item;
+			item = tmp;
+		}
+	}	
+}
+int main()
+{
+	int size = 6;
+	int *numbers = malloc(size*sizeof(int));
```
Comment (christianbender) [inline sorting/cycle_sort.c]: Deallcoate this space at the end of the main-function

File: sorting/cycle_sort.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+				if (numbers[i] < item)
+					pos++;
+			while (item == numbers[pos])
+				pos++;
+			int tmp = numbers[pos];
+			numbers[pos] = item;
+			item = tmp;
+		}
+	}	
+}
```
Comment (christianbender) [inline sorting/cycle_sort.c]: Newline after this line

Comment (gbrunofranco) [reply sorting/cycle_sort.c]: Fixed.

**Pull request closed without merging**